service/backend/pchain/indexer/parser.go

```diff
@@ -110,14 +113,14 @@ func (p *parser) GetGenesisBlock(ctx context.Context) (*ParsedGenesisBlock, erro
 	genesisTxs = append(genesisTxs, genesisState.Validators...)
 	genesisTxs = append(genesisTxs, genesisState.Chains...)
 
-	// Genesis commit block's parent ID is the hash of genesis state
-	var genesisParentID ids.ID = hashing.ComputeHash256Array(bytes)
-
-	// Genesis Block is not indexed by the indexer, but its block ID can be accessed from block 0's parent id
-	genesisChildBlock, err := p.parseBlockAtHeight(ctx, 1)
+	// Build genesis ID and ParentID as it's done in platformVM'State,
+	// without polling indexer (for backward compatibility).
+	genesisParentID := hashing.ComputeHash256Array(bytes)
+	genesisBlock, err := pBlocks.NewApricotCommitBlock(genesisParentID, 0)
```

Comment:
this is how it's done in platformVM. See https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/state/state.go#L1453:L1457